core/src/main/java/org/apache/carbondata/core/dictionary/client/DictionaryClient.java

```diff
@@ -86,7 +78,11 @@ public DictionaryKey getDictionary(DictionaryKey key) {
    * shutdown dictionary client
    */
   public void shutDown() {
-    clientBootstrap.releaseExternalResources();
-    clientBootstrap.shutdown();
+    workerGroup.shutdownGracefully();
```

Comment:
Method `clientBootstrap.group()` do not return `NioEventLoopGroup` , so we cannot shutdown this way